src/function/list/list_slice_function.cpp

```diff
@@ -9,23 +9,22 @@ namespace kuzu {
 namespace function {
 
 static void normalizeIndices(int64_t& startIdx, int64_t& endIdx, uint64_t size) {
+    int64_t sSize = size;
     if (startIdx < 0) {
         startIdx = size + startIdx + 1;
     }
-    if (endIdx <= 0) {
-        endIdx = size + endIdx + 1;
-    }
-
     if (startIdx <= 0) {
         startIdx = 1;
     }
-
-    if ((uint64_t)endIdx > size) {
-        endIdx = size + 1;
+    if (endIdx < 0) {
+        endIdx = size + endIdx + 1;
     }
-
-    if (startIdx > endIdx) {
-        endIdx = startIdx;
+    if (endIdx > sSize) {
+        endIdx = size;
+    }
+    if (endIdx < startIdx || startIdx > sSize) {
```

Comment:
I believe you don't need or condition. L22 already guarantees `endIdx <= size`. If `startIdx > size` then it must also satisfy `startIdx > endIdx`. So just keep the first condition (or we should the original check)